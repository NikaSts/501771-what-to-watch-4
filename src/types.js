import {string, number, shape, arrayOf, bool} from 'prop-types';

export const movieType = shape({
  id: number.isRequired,
  title: string.isRequired,
  runTime: number.isRequired,
  genre: string.isRequired,
  releaseDate: number.isRequired,
  poster: string.isRequired,
  previewImage: string.isRequired,
  backgroundImage: string.isRequired,
  backgroundColor: string.isRequired,
  ratingScore: number.isRequired,
  ratingLevel: string.isRequired,
  ratingCount: number.isRequired,
  paragraphs: arrayOf(string).isRequired,
  director: string.isRequired,
  stars: arrayOf(string).isRequired,
  preview: string.isRequired,
  isFavorite: bool.isRequired,
});

export const promoMovieType = shape({
  id: number.isRequired,
  title: string.isRequired,
  genre: string.isRequired,
  releaseDate: number.isRequired,
  poster: string.isRequired,
  backgroundImage: string.isRequired,
  backgroundColor: string.isRequired,
});

export const cardMovieType = shape({
  title: string.isRequired,
  image: string.isRequired,
});
