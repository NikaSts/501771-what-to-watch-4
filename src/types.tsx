export interface MovieType {
  id: number;
  title: string;
  runTime: number;
  genre: string;
  releaseDate: number;
  poster: string;
  previewImage: string;
  backgroundImage: string;
  backgroundColor: string;
  ratingScore: number;
  ratingLevel?: string;
  ratingCount: number;
  description: string;
  director: string;
  stars: Array<string>;
  preview: string;
  video: string;
  isFavorite: boolean;
}

export interface UserDataType {
  id: number;
  name: string;
  email: string;
  avatar: string;
}

export interface ReviewType {
  id: number;
  user: {
    id: number;
    name: string;
  };
  rating: number;
  comment: string;
  date: string;
}
