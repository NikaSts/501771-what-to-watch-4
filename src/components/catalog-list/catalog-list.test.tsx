import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import history from '../../history';
import CatalogList from './catalog-list';
import {MovieType} from '../../types';

const movies: Array<MovieType> = [
  {
    id: 3,
    title: `First Movie`,
    runTime: 1,
    genre: ``,
    releaseDate: 1,
    poster: ``,
    previewImage: ``,
    backgroundImage: ``,
    backgroundColor: ``,
    ratingScore: 1,
    ratingLevel: ``,
    ratingCount: 1,
    description: ``,
    director: ``,
    stars: [``],
    preview: ``,
    video: ``,
    isFavorite: false,
  },
  {
    id: 4,
    title: `Second Movie`,
    runTime: 1,
    genre: ``,
    releaseDate: 1,
    poster: ``,
    previewImage: ``,
    backgroundImage: ``,
    backgroundColor: ``,
    ratingScore: 1,
    ratingLevel: ``,
    ratingCount: 1,
    description: ``,
    director: ``,
    stars: [``],
    preview: ``,
    video: ``,
    isFavorite: false,
  },
];

it(`CatalogList should render correctly`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <CatalogList
            movies={movies}
          />
        </Router>
    )
      .toJSON();

  expect(tree).toMatchSnapshot();
});
