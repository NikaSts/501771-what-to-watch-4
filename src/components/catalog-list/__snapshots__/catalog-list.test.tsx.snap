// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogList should render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      href="/films/3"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          preload="none"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/3"
        onClick={[Function]}
      >
        First Movie
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      href="/films/4"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          preload="none"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/4"
        onClick={[Function]}
      >
        Second Movie
      </a>
    </h3>
  </article>
</div>
`;
